src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.TypeInit.cs

```diff
@@ -15,6 +15,13 @@ namespace ILCompiler
     // Manages policies around static constructors (.cctors) and static data initialization.
     partial class CompilerTypeSystemContext
     {
+        private ExplicitClassConstructors _explicitClassConstructors;
```

Comment:
We could change the list of types in `ExplicitClassConstructors` to each have no .cctor under CoreRT, replaced with an `Initialize` method. Under .NET Native, the .cctor would call Initialize().  In CoreRT startup code, we still need a list of these things to call since we can't refer to types outside S.P.C in StartupCodeHelpers.  I guess now that list wouldn't need to be passed into the TypeSystemContext so that's a win.

I'll add that I don't really like that pattern though, adding CoreRT init cruft into the class library feels dirtier than special handling all restricted to the compiler.